tss-esapi/src/context.rs

```diff
@@ -107,13 +107,19 @@ impl Context {
         )?;
 
         let esys_context = unsafe { Some(Malloced::from_raw(esys_context)) };
-        Ok(Context {
+        let mut context = Context {
             esys_context,
             sessions: (None, None, None),
             _tcti_context,
             handle_manager: HandleManager::new(),
             cached_tpm_properties: HashMap::new(),
-        })
+        };
+
+        if matches!(tcti_name_conf, TctiNameConf::LibTpms { .. }) {
```

Comment:
If this comparison is moved before `initialize` and set to some variable (say `needs_clear_startup`) then I guess we could save one clone call.